src/telemetry/ProcessStats.cc

```diff
@@ -143,9 +156,7 @@ process_stats get_process_stats() {
         result.vms = vmsize_bytes;
         result.cpu = static_cast<double>(utime_ticks + stime_ticks) / ticks_per_second;
 
-        zeek::filesystem::path fd_path{"/proc/self/fd"};
-        result.fds =
```

Comment:
I'd love to understand where this actually goes haywire in the `fd` directory — it's a pretty amazing failure mode! It's a pity because this looks so much nicer.

Reply:
Same. The only thing I can think is that either we get into a mode where the `directory_iterator` is creating file descriptors as it loops, or the `opendir`/`readdir` version caches the current list of file descriptors before we start looping so it's always over a fixed set.